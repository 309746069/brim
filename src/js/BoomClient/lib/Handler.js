/* @flow */

export default class Handler {
  abortFunc: Function
  channelCallbacks: {[number]: Function[]}
  doneCallbacks: Function[]
  eachCallbacks: Function[]
  errorCallbacks: Function[]
  abortCallbacks: Function[]
  isDone: boolean
  isAborted: boolean

  constructor(abortFunc: Function) {
    this.abortFunc = abortFunc
    this.isDone = false
    this.isAborted = false
    this.channelCallbacks = {}
    this.doneCallbacks = []
    this.eachCallbacks = []
    this.errorCallbacks = []
    this.abortCallbacks = []
  }

  abortRequest() {
    if (!this.isDone) {
      this.abortFunc()
    }
  }

  abort(cb: Function) {
    this.abortCallbacks.push(cb)
    return this
  }

  each(callback: Function) {
    this.eachCallbacks.push(callback)
    return this
  }

  channel(id: number, callback: Function) {
    if (this.channelCallbacks[id]) {
      this.channelCallbacks[id].push(callback)
    } else {
      this.channelCallbacks[id] = [callback]
    }
    return this
  }

  done(callback: Function) {
    this.doneCallbacks.push(callback)
    return this
  }

  error(callback: Function) {
    this.errorCallbacks.push(callback)
    return this
  }

  onError(error: *) {
    if (this.isAborted) return

    this.errorCallbacks.forEach(cb => cb(error))
  }

  onDone(payload: *) {
    if (this.isAborted) return

    this.isDone = true
    this.doneCallbacks.forEach(cb => cb(payload))
  }

  onAbort() {
    this.isAborted = true
    this.abortCallbacks.forEach(cb => cb())
  }

  receive(payload: *) {
    if (this.isAborted) return

    this.eachCallbacks.forEach(cb => cb(payload))

    if ("channel_id" in payload) {
      this.channelCallback(payload.channel_id, payload)
    }
  }

  channelCallback(id: number, payload: *) {
    if (id in this.channelCallbacks) {
      this.channelCallbacks[id].forEach(cb => cb(payload))
    }
  }
}
