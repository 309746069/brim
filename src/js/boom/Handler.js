export default class Handler {
  constructor() {
    this.channelCallbacks = {}
    this.doneCallbacks = []
    this.eachCallbacks = []
    this.errorCallbacks = []
  }

  each(callback) {
    this.eachCallbacks.push(callback)
    return this
  }

  channel(id, callback) {
    this.channelCallbacks[id] = callback
    return this
  }

  done(callback) {
    this.doneCallbacks.push(callback)
    return this
  }

  error(callback) {
    this.errorCallbacks.push(callback)
    return this
  }

  onError(error) {
    this.errorCallbacks.forEach(cb => cb(error))
  }

  onDone(payload) {
    this.doneCallbacks.forEach(cb => cb(payload))
  }

  receive(payload) {
    this.eachCallbacks.forEach(cb => cb(payload))

    if ("channel_id" in payload) {
      this.channelCallback(payload.channel_id, payload)
    }
  }

  channelCallback(id, payload) {
    if (id in this.channelCallbacks) this.channelCallbacks[id](payload)
  }
}
