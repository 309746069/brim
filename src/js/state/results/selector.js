/* @flow */
import type {State} from "../reducers/types"

export function getResults(state: State) {
  return state.results
}
