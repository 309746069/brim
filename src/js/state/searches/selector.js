/* @flow */

import type {State} from "../reducers/types"

export function getSearches(state: State) {
  return state.searches
}

export function getSearch(state: State, name: string) {
  return state.searches[name]
}
