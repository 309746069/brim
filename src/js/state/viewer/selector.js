/* @flow */
import type {State} from "../reducers/types"

export function getViewerLogs(state: State) {
  return state.viewer.logs
}

export function getViewerStatus(state: State) {
  return state.viewer.status
}

export function getViewerColumns(state: State) {
  return state.viewer.columns
}
