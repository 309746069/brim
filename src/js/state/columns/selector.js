/* @flow */
import {createSelector} from "reselect"

import type {ColumnsState} from "./types"
import type {State} from "../types"
import type {ViewerColumns} from "../viewer/types"
import {getViewerColumns} from "../viewer/selector"
import {uniqBy} from "../../lib/Array"
import TableColumns from "../../models/TableColumns"
import columnKey from "../../lib/columnKey"

export function getColumns(state: State) {
  return state.columns
}

export const getCurrentTableColumns = createSelector<
  State,
  void,
  TableColumns,
  ViewerColumns,
  ColumnsState,
  *
>(
  getViewerColumns,
  getColumns,
  (viewerColumns, columnSettings) => {
    let tableKey = getTableKey(viewerColumns)
    let columns = getUniqColumns(viewerColumns)
    return new TableColumns(tableKey, columns, columnSettings[tableKey])
  }
)

function getTableKey(columns) {
  let ids = Object.keys(columns)
  if (ids.length === 0) return "none"
  if (ids.length === 1) return ids[0]
  return "temp"
}

function getUniqColumns(columns) {
  let allCols = []
  for (const id in columns) allCols = [...allCols, ...columns[id]]
  return uniqBy(allCols, columnKey)
}
