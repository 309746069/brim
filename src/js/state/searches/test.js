/* @flow */

import {
  clearSearches,
  registerSearch,
  setSearchStats,
  setSearchStatus
} from "./actions"
import {getSearch, getSearches} from "./selector"
import Handler from "../../BoomClient/lib/Handler"
import initTestStore from "../../test/initTestStore"

describe("searches reducer", () => {
  let store

  beforeEach(() => {
    store = initTestStore()
    store.dispatchAll([
      registerSearch("Histogram", {tag: "viewer", handler: new Handler()}),
      registerSearch("Logs", {tag: "detail", handler: new Handler()})
    ])
  })

  test("#registerSearch", () => {
    let searches = getSearches(store.getState())

    expect(Object.keys(searches)).toHaveLength(2)
  })

  test("#registerSearch adds a tag", () => {
    let search = getSearch(store.getState(), "Histogram")

    expect(search.tag).toBe("viewer")
  })

  test("#setSearchStatus", () => {
    store.dispatch(setSearchStatus("Histogram", "SUCCESS"))
    let search = getSearch(store.getState(), "Histogram")

    expect(search.status).toBe("SUCCESS")
  })

  test("#setSearchStatus for search that does not exist", () => {
    expect(() => {
      store.dispatch(setSearchStatus("NoSearchHere", "SUCCESS"))
    }).toThrowErrorMatchingSnapshot()
  })

  test("#setSearchStats", () => {
    let stats = {
      tuplesRead: 100,
      tuplesMatched: 100,
      bytesMatched: 100,
      bytesRead: 2300,
      startTime: 1,
      updateTime: 3
    }

    store.dispatch(setSearchStats("Histogram", stats))
    let search = getSearch(store.getState(), "Histogram")

    expect(search.stats).toEqual(stats)
  })

  test("#setSearchStats for search that does not exist", () => {
    expect(() => {
      store.dispatch(
        setSearchStats("NoSearchHere", {
          tuplesRead: 100,
          tuplesMatched: 100,
          bytesMatched: 100,
          bytesRead: 2300,
          startTime: 1,
          updateTime: 3
        })
      )
    }).toThrowErrorMatchingSnapshot()
  })

  test("searches clear", () => {
    store.dispatch(clearSearches())

    expect(getSearches(store.getState())).toEqual({})
  })

  test("searches clear with tag", () => {
    store.dispatch(clearSearches("viewer"))
    let searches = getSearches(store.getState())

    expect(Object.keys(searches)).toHaveLength(1)
  })
})
