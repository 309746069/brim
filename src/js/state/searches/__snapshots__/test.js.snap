// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searches reducer #setSearchStats for search that does not exist 1`] = `"Trying to update search that does not exist: NoSearchHere"`;

exports[`searches reducer #setSearchStatus for search that does not exist 1`] = `"Trying to update search that does not exist: NoSearchHere"`;
